src/internal/api/authenticator_test.go

```diff
@@ -208,9 +211,20 @@ func TestAccessToken(t *testing.T) {
 			// mock httpClient
 			httpClient := &http.Client{Timeout: 3 * time.Second}
 			client := NewHTTPClient(cfgSrv, httpClient, server.URL)
-			auth := authenticator{http: client, cfgSvc: config.NewService(&storage), backoffCfg: backoffCfg{
-				status: v.authStatus,
-			}}
+			auth := authenticator{
+				http:   client,
+				cfgSvc: config.NewService(&storage),
+				mqtt: fimpgo.NewMqttTransport(
+					cfg.MQTTServerURI,
+					cfg.MQTTClientIDPrefix,
+					cfg.MQTTUsername,
+					cfg.MQTTPassword,
+					true,
+					1,
+					1,
+				),
+				backoff: backoff.NewStateful(0, 0, 0, 0, 0),
+			}
 
 			token, err := auth.AccessToken()
 			if err != nil {
```

Comment:
This logic is incorrect. If `v.errorContains` is not equal to an empty string, we should check whether an error is not nil and contains the expected message. Otherwise we should assert if an error is nil.